content/docs/pipelines/sdk/python-based-visualizations.md

```diff
@@ -4,174 +4,59 @@ description = "Predefined and arbitrary visualizations of pipeline outputs"
-              # a visualization is generated here.
-              pass
-            else:
-              # Variable of name key is not provided.
-              # You can provide a default operation here if a variable is not
-              # provided or throw an error if the variable must be provided.
-              pass
-            ...
-            # Accessing a variable
-            key = variables["key"]
-            # or
-            key = variables.get("key")
-            ...
-            ```
-            * Additional details about how this is implemented can be found
-            [here](https://github.com/kubeflow/pipelines/blob/master/backend/
-            src/apiserver/visualization/exporter.py#L93).
-9. Add any new dependencies to the **requirements.txt** file in the
-`backend/src/apiserver/visualization` directory.
-    * This only needs to be updated if new dependencies are added.
-    * If new dependencies are added, the
-    [**third_party_licenses.csv**](https://github.com/kubeflow/pipelines/blob/
-    master/backend/src/apiserver/visualization/third_party_licenses.csv) file
-    must also be updated.
-        
-        ```csv
-        package_name,url_to_package_license,license_name
-        ```
-        * `package_name`: the name of the package on [pypi](https://pypi.org/)
-        * `url_to_package_license`: the url where the license of the package can
-        be downloaded from
-        * `license_name`: name of package license
-        * Examples for all the columns can be found [here](https://github.com/
-        kubeflow/pipelines/blob/master/backend/src/apiserver/visualization/
-        third_party_licenses.csv).
-10. Submit these changes as a PR or build docker image for usage within your
-cluster.
-
-## Using Predefined Visualizations
-
-1. Open the details of a run
-2. Select a component
+1. Open the details of a run.
+2. Select a component.
     * The component that is selected does not matter. But, if you want to
     visualize the output of a specific component, it is easier to do that within
     that component.
-3. Select the **Artifacts** tab
-4. At the top of the tab you should see a card named **Visualization Creator**
+3. Select the **Artifacts** tab.
+4. At the top of the tab you should see a card named **Visualization Creator**.
 5. Within the card, provide a visualization type, a source, and any necessary
-arguments
-    * Any required or optional arguments will be shown as a placeholder
-6. Click **Generate Visualization**
-7. View generated visualization by scrolling down
+arguments.
+    * Any required or optional arguments will be shown as a placeholder.
+6. Click **Generate Visualization**.
+7. View generated visualization by scrolling down.
 
-## Using Arbitrary Visualizations
+## Using arbitrary visualizations
 
-1. Enable arbitrary visualizations within Kubeflow Pipelines
-2. Open the details of a run
-3. Select a component
+1. Enable arbitrary visualizations within Kubeflow Pipelines.
```

Comment:
Thanks for your earlier reply! Just noting a TODO: Update this list item to include details on how to enable arbitrary visualizations once the way to pass variables from the cluster to the UI is determined